torch/utils/_pytree.py

```diff
@@ -136,12 +139,12 @@ def _register_pytree_node(
             "be None or registered."
         )
 
-    type_fqn = f"{typ.__module__}.{typ.__name__}"
-    serialize_node_def = _SerializeNodeDef(
-        typ, type_fqn, to_dumpable_context, from_dumpable_context
-    )
-    SUPPORTED_SERIALIZED_TYPES[typ] = serialize_node_def
-    SERIALIZED_TYPE_TO_PYTHON_TYPE[type_fqn] = typ
+    if type_fqn is not None:
+        serialize_node_def = _SerializeNodeDef(
+            typ, type_fqn, to_dumpable_context, from_dumpable_context
+        )
+        SUPPORTED_SERIALIZED_TYPES[typ] = serialize_node_def
+        SERIALIZED_TYPE_TO_PYTHON_TYPE[type_fqn] = typ
```

Comment:
What should the behavior be on duplicate registrations? Right now it is "last one wins", but perhaps we should explicitly error?

Reply:
I think in the near future they will disable duplicate registrations: https://github.com/pytorch/pytorch/pull/112111/files#r1376411179